File: General problems/02_palindrome/palindrome.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	fmt.Println(palindrome("naman"))
	fmt.Println(palindrome("A man a plan a canal Panama"))
	fmt.Println(palindrome("A man a plan a canal Panam"))
	fmt.Println(palindrome("Ronald"))
}

// naman
func palindrome(s string) bool {
	s = strings.ToLower(strings.Join(strings.Fields(s), ""))
	for i := 0; i < len(s)/2; i++ {
		if s[i] != s[len(s)-1-i] {
			return false
		}
	}
	return true
}

File: General problems/02_palindrome/02_python_palindrome/palindrome.py

def palindrome(s):
    s = s.lower()
    for i in range (len(s)//2):
        if s[i]!= s[len(s)-1-i]:
            return False 
    return True
print(palindrome("naman"))    
print(palindrome("A man a plan a canal Panama")) 